components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/NotificationTemplateManagerImpl.java

```diff
@@ -0,0 +1,720 @@
+                                                  String tenantDomain)
+            throws NotificationTemplateManagerException {
+
+        try {
+            return unifiedTemplatePersistenceManager.isNotificationTemplateTypeExists(templateTypeDisplayName,
+                    notificationChannel, tenantDomain);
+        } catch (NotificationTemplateManagerServerException e) {
+            String error = String.format("Error when retrieving templates of %s tenant.", tenantDomain);
+            throw handleServerException(error, e);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<NotificationTemplate> getAllNotificationTemplates(String notificationChannel, String tenantDomain)
+            throws NotificationTemplateManagerException {
+
+        try {
+            if (OrganizationManagementUtil.isOrganization(tenantDomain)) {
+                // Return the root organization's email templates.
+                tenantDomain = getRootOrgTenantDomain(tenantDomain);
+            }
+            return userDefinedTemplatePersistenceManager.listAllNotificationTemplates(
+                    notificationChannel, tenantDomain);
+        } catch (NotificationTemplateManagerServerException e) {
+            String error = String.format("Error when retrieving templates of %s tenant.", tenantDomain);
+            throw handleServerException(error, e);
+        } catch (OrganizationManagementException e) {
+            throw handleServerException(e.getMessage(), e);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<NotificationTemplate> getNotificationTemplatesOfType(String notificationChannel,
+                                                                     String templateDisplayName, String tenantDomain)
+            throws NotificationTemplateManagerException {
+
+        return getNotificationTemplatesOfType(notificationChannel, templateDisplayName, tenantDomain, null);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<NotificationTemplate> getNotificationTemplatesOfType(String notificationChannel,
+                                              String templateDisplayName, String tenantDomain, String applicationUuid)
+            throws NotificationTemplateManagerException {
+
+        try {
+            if (OrganizationManagementUtil.isOrganization(tenantDomain)) {
+                // Return the root organization's email templates.
+                tenantDomain = getRootOrgTenantDomain(tenantDomain);
+            }
+            assertTemplateTypeExists(templateDisplayName, notificationChannel, tenantDomain);
+            List<NotificationTemplate> notificationTemplates =
+                    userDefinedTemplatePersistenceManager.listNotificationTemplates(templateDisplayName,
+                            notificationChannel, applicationUuid, tenantDomain);
+            if (notificationTemplates == null) {
```

Comment:
if not returning a null remove this null check.

Reply:
addressed with https://github.com/wso2-extensions/identity-event-handler-notification/pull/250/commits/4a4dae818d530081e8a91a33d6d70da13c375c9c